includes/class-wc-payments-utils.php

```diff
@@ -149,51 +149,62 @@ public static function mark_payment_expired( $order ) {
 	}
 
 	/**
-	 * Returns a charge_id for an "Order #" search term.
+	 * Returns the charge_id for an "Order #" search term
+	 * or all charge_ids for a "Subscription #" search term.
 	 *
 	 * @param string $term Search term.
 	 *
-	 * @return string The charge_id for the order, or empty if no order is found.
+	 * @return array|null The charge_id(s) for the order or subscription, or null if no charges are found.
 	 */
-	public static function get_charge_id_from_search_term( $term ) {
-		$order_term = 'Order #';
+	public static function get_charge_ids_from_search_term( $term ) {
+		$order_term = __( 'Order #', 'woocommerce-payments' );
 		if ( substr( $term, 0, strlen( $order_term ) ) === $order_term ) {
 			$term_parts = explode( '#', $term, 2 );
 			$order_id   = isset( $term_parts[1] ) ? $term_parts[1] : '';
+			$order      = wc_get_order( $order_id );
+			if ( $order ) {
+				return [ $order->get_meta( '_charge_id' ) ];
+			}
 		}
-		if ( ! isset( $order_id ) || empty( $order_id ) ) {
-			return '';
-		}
-
-		$order = wc_get_order( $order_id );
 
-		if ( ! $order ) {
-			return '';
+		$subscription_term = __( 'Subscription #', 'woocommerce-payments' );
+		if ( function_exists( 'wcs_get_subscription' ) && substr( $term, 0, strlen( $subscription_term ) ) === $subscription_term ) {
+			$term_parts      = explode( '#', $term, 2 );
+			$subscription_id = isset( $term_parts[1] ) ? $term_parts[1] : '';
+			$subscription    = wcs_get_subscription( $subscription_id );
+			if ( $subscription ) {
+				return array_map(
+					function ( $order ) {
+						return $order->get_meta( '_charge_id' );
+					},
+					$subscription->get_related_orders( 'all' )
+				);
+			}
 		}
 
-		return $order->get_meta( '_charge_id' );
+		return null;
```

Comment:
Do you think we could opt for an empty array instead? The simplifies the function to one type.

Reply:
👍 Done in https://github.com/Automattic/woocommerce-payments/pull/936/commits/eb3823d630870d2a55c6c51683ffd77cbfc50cd4